frontend/src/bundles/common/components/slider/slider.tsx

```diff
@@ -0,0 +1,62 @@
+import { Box, Slider } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { useCallback } from 'react';
+
+import { getValidClassNames } from '../../helpers/helpers.js';
+import classes from './style.module.css';
+import { defaultSliderStyle } from './styles.js';
+
+type Option = {
+    label?: string;
+    value: number;
+};
+
+type Properties = {
+    marks: Option[];
+    label?: string;
+    value?: number;
+    className?: string;
+    step?: number | null;
+    valueLabelDisplay?: 'auto' | 'on' | 'off';
+};
+
+const CustomSlider: React.FC<Properties> = ({
+    marks = [],
+    label,
+    value,
+    className = classes.sliderContainerStyle,
+    step = null,
+    valueLabelDisplay = 'on',
+}) => {
+    const StyledSlider = styled(Slider)(defaultSliderStyle);
+
+    const getValueLabel = useCallback(
+        (value: number): string | null => {
+            const label = marks.find((mark) => mark.value === value)?.label;
+            return label ?? null;
```

Comment:
This variant much better

```suggestion
            const mark = marks.find((mark) => mark.value === value);
            
            return mark?.label ?? null;
```

Reply:
ok, I have tested it, and can add